manager/drivers/secrets.go

```diff
@@ -26,12 +27,23 @@ func NewSecretDriver(plugin plugingetter.CompatPlugin) *SecretDriver {
 }
 
 // Get gets a secret from the secret provider
-func (d *SecretDriver) Get(spec *api.SecretSpec) ([]byte, error) {
+func (d *SecretDriver) Get(spec *api.SecretSpec, serviceSpec *swarm.ServiceSpec) ([]byte, error) {
```

Comment:
Why does this take a `swarm.ServiceSpec` as defined in `github.com/docker/docker/api/types/swarm` instead of the native swarmkit type `api.ServiceSpec`?

I suggested vendoring and using the `convert` package back when we were considering sending the whole service spec in JSON format, but if we are only sending selected fields with our own JSON schema, I don't see the advantage of involving the types defined in `docker/docker`.